### PR TITLE
Fixed Issue #3: Create Sidebar Component

### LINKED ISSUE #3: Create sidebar component
Create a React component for the sidebar.

The component should take the following props:
- An array of names of subscribed streams (we'll implement unread indicator later).
- The user's name (to show profile)

Right now, don't worry about what happens on clicking. Just make clicking on any stream or button do a `console.log('Clicked!')` :)

### PR DESCRIPTION
**Pull request by NupurJ**
Resolves #3. Includes a Hamburger, which, when clicked, displays the sidebar. The props to be passed to the sidebar (username and streams list) have been added in a variable in src/index.js for now.

image

### CODE REVIEW

File: src/components/sidebar/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      {props.streamName}
+    </ListItem>
+  );
+}
+
+function StreamsList(props) {
+  let list = new Array();
+  for (let stream of props.streams) {
+    list.push(<StreamItem streamName={stream} />);
+  }
```
Comment (yashsriv) [inline src/components/sidebar/index.js:30]: This is generally considered a better way of looping over arrays in JS - clear and succint:

```suggestion
  const list = props.stream.map(stream => <StreamItem streamName={stream} />);
```

File: src/index.js [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
+  "And It Goes Like This",
+];
 ReactDOM.render(
   <React.StrictMode>
-    <App />
+    <Home name={name} streams={streams} />
   </React.StrictMode>,
-  document.getElementById('root')
+  document.getElementById("root")
 );
```
Comment (NilayMajorwar) [inline src/index.js:20]: You can remove the Home component from here btw, let it be `<App />`. We'll add a higher-order component later that manages the UI.

**Comment by NilayMajorwar** [comment]
Nice work! Although there are some more things:
- We're working on the web app right now, so it's better to have a larger, fixed-width sidebar that is open by default.
- Let's keep the profile section at the bottom of the sidebar (since it's not going to be opened very often by the user), and add a new section for the app name at the top of the sidebar. Use a divider between the app name section and the streams list.

**Comment by NupurJ** [comment]
Done!

**Pull request closed without merging**